api/app/Policies/PoolPolicy.php

```diff
@@ -71,26 +72,19 @@ public function viewAnyPublished(?User $user)
      */
     public function create(User $user, $request)
     {
-        $team_id = null;
         if (array_key_exists('team_id', $request)) {
             $team_id = $request['team_id'];
-        }
-
-        // Failed to get the new pools team - reject
-        // This should only occur if the mutation structure has changed
-        if (is_null($team_id)) {
-            return Response::deny('Pool must be associated with a team when it is created.');
-        }
 
-        // Get the team to check against
-        $team = Team::find($team_id);
+            // Get the team to check against
+            $team = Team::find($team_id);
 
-        // Confirm the user can create pools for the team
-        if ($user->isAbleTo('create-team-pool', $team)) {
-            return true;
-        } else {
-            return Response::deny('Cannot create a pool for that team.');
+            // Confirm the user can create pools for the team
+            if ($user->isAbleTo('create-team-pool', $team)) {
+                return true;
+            }
         }
+
+        return Response::deny('Cannot create a pool for that team.');
```

Comment:
I like that previously the error response was different between 'Pool must be associated with a team when it is created.' and 'Cannot create a pool for that team.'. In fact, I'd recommend throwing one other response if $team is null, something like "Cannot find a team matching team_id".

Reply:
I've added some error responses (back) to this now.